instance-scheduler/main.go

```diff
@@ -268,7 +277,15 @@ func handler(ctx context.Context, request InstanceSchedulingRequest) (events.API
 		log.Fatal(err)
 	}
 
-	accounts := getNonProductionAccounts(cfg, skipAccounts)
+	// Get the secret ID (ARN) of the environment management secret from the parameter store
+	ssmClient := ssm.NewFromConfig(cfg)
```

Comment:
Is moving the retrieval of the param and secret out of the getNonProdAccounts function make it easier to mock/test?

Reply:
Yes. It's a bit of a domino effect. It is impossible to test functions if you define clients inside them (as you cannot mock them), hence getSecret and getParameter now accept clients. Now, I could pass clients into the getNonProdAccounts accounts and then define/assign secretId and environments in there, but that means in order to test getNonProdAccounts, you would need to mock two objects, instead you do not need to mock anything, so less work.

Reply:
Makes sense.